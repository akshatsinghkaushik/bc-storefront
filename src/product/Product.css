.product-container {
  margin: 2rem 0 0 0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.product-content {
  margin: 3rem 0 0 0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
}

.product-details {
  margin: 0 0 0 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.product-image {
  width: 40rem;
  object-fit: cover;
}

.product-brand,
.product-price,
.product-title,
.product-description {
  margin: 0 0 2rem 0;
}

.product-brand {
  color: gray;
  font-size: 15px;
  font-weight: 600;
}

.product-title {
  font-size: 35px;
  font-weight: 600;
  width: 22rem;
  text-align: center;
}

.product-price {
  color: rgb(153, 153, 153);
  font-size: 20px;
  font-weight: 600;
}

.product-description {
  color: rgb(109, 108, 108);
  width: 25rem;
}

.product-quantity {
  margin: 2rem 0 0 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.add-to-cart {
  margin-left: 1rem;
  background: none;
  background-color: black;
  color: white;
  border: none;
  padding: 1rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}
