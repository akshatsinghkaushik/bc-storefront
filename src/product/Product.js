import React, { useState, useContext } from 'react';
import './Product.css';
import CartContext from '../context/CartContext';

const Product = ({ match, testParam }) => {
  const [quantity, setQuantity] = useState(1);
  const cartContext = useContext(CartContext);

  let product;
  if (match !== undefined) {
    product = cartContext.getProduct(match.params.id);
  } else {
    product = cartContext.getProduct(testParam);
  }

  return (
    <div className="Product">
      {/* {match.params.id} */}
      <div className="product-container">
        <h5 className="product-location">
          HOME/PLATES/
          <span style={{ color: 'gray' }}>{product.title.toUpperCase()}</span>
        </h5>
        <div className="product-content">
          <img
            className="product-image"
            src={`/media/${product.image}`}
            alt={product.title}
          />
          <div className="product-details">
            <div className="product-brand">{product.brand}</div>
            <div className="product-title">{product.title}</div>
            <div className="product-price">
              $ {Number(product.price).toFixed(2)}
            </div>
            <div className="product-description">{product.description}</div>
            <hr style={{ height: '1px', width: '100%', margin: '0' }} />
            <div className="product-quantity">
              <div className="total-amount">{quantity}</div>
              <div className="incre-decre-buttons">
                <button
                  className="incre-button"
                  onClick={() => setQuantity(quantity + 1)}
                >
                  +
                </button>
                <button
                  onClick={() => {
                    if (quantity != 1) setQuantity(quantity - 1);
                  }}
                >
                  -
                </button>
              </div>
              <button
                className="add-to-cart"
                onClick={() => cartContext.addItemToCart(product, quantity)}
              >
                ADD TO CART
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Product;
