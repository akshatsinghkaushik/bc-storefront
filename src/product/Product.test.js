import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Product from './Product';
import MockCartProvider from '../tests/MockCartProvider';

it('Cart Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <Product testParam={'Blue Stripe Stoneware Plate'} />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
