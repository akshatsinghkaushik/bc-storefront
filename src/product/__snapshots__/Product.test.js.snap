// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Renders correctly 1`] = `
<div
  className="Product"
>
  <div
    className="product-container"
  >
    <h5
      className="product-location"
    >
      HOME/PLATES/
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
        BLUE STRIPE STONEWARE PLATE
      </span>
    </h5>
    <div
      className="product-content"
    >
      <img
        alt="Blue Stripe Stoneware Plate"
        className="product-image"
        src="/media/blue-stripe-stoneware-plate.jpg"
      />
      <div
        className="product-details"
      >
        <div
          className="product-brand"
        >
          Kiriko
        </div>
        <div
          className="product-title"
        >
          Blue Stripe Stoneware Plate
        </div>
        <div
          className="product-price"
        >
          $ 
          40.00
        </div>
        <div
          className="product-description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.
        </div>
        <hr
          style={
            Object {
              "height": "1px",
              "margin": "0",
              "width": "100%",
            }
          }
        />
        <div
          className="product-quantity"
        >
          <div
            className="total-amount"
          >
            1
          </div>
          <div
            className="incre-decre-buttons"
          >
            <button
              className="incre-button"
              onClick={[Function]}
            >
              +
            </button>
            <button
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <button
            className="add-to-cart"
            onClick={[Function]}
          >
            ADD TO CART
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
